# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpfe/appshell/nsAppShellService.cpp
# Commit: f144394b822c
# Full Hash: f144394b822cea0388dc4cbc0241f17fe2e3faa3
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-08 03:36:50
# Regressor Bug: 1616353
# File Overlap Count: 1
# Description:
#   Bug 1616353 - Part 4: Don't second-guess CHROME_{REMOTE,FISSION}_WINDOW flags, r=kmag
#   
#   This could cause issues, as we could incorrectly override the decision made by
#   nsWindowWatcher, and then override the decision again after the method returns.
#   This could cause the initial content browser to be configured incorrectly.
# ==============================================================================

diff -r 9bae7eb4e34a -r f144394b822c xpfe/appshell/nsAppShellService.cpp
--- a/xpfe/appshell/nsAppShellService.cpp	Tue Apr 07 21:39:04 2020 +0000
+++ b/xpfe/appshell/nsAppShellService.cpp	Tue Apr 07 21:39:07 2020 +0000
@@ -685,20 +685,11 @@
   // Enforce the Private Browsing autoStart pref first.
   bool isPrivateBrowsingWindow =
       Preferences::GetBool("browser.privatebrowsing.autostart");
-  bool isUsingRemoteTabs = mozilla::BrowserTabsRemoteAutostart();
-  bool isUsingRemoteSubframes = StaticPrefs::fission_autostart();
 
   if (aChromeMask & nsIWebBrowserChrome::CHROME_PRIVATE_WINDOW) {
     // Caller requested a private window
     isPrivateBrowsingWindow = true;
   }
-  if (aChromeMask & nsIWebBrowserChrome::CHROME_REMOTE_WINDOW) {
-    isUsingRemoteTabs = true;
-  }
-  if (aChromeMask & nsIWebBrowserChrome::CHROME_FISSION_WINDOW) {
-    isUsingRemoteSubframes = true;
-  }
-
   nsCOMPtr<mozIDOMWindowProxy> domWin = do_GetInterface(aParent);
   nsCOMPtr<nsIWebNavigation> webNav = do_GetInterface(domWin);
   nsCOMPtr<nsILoadContext> parentContext = do_QueryInterface(webNav);
@@ -710,10 +701,6 @@
     isPrivateBrowsingWindow = parentContext->UsePrivateBrowsing();
   }
 
-  if (parentContext) {
-    isUsingRemoteTabs = parentContext->UseRemoteTabs();
-    isUsingRemoteSubframes = parentContext->UseRemoteSubframes();
-  }
 
   nsCOMPtr<mozIDOMWindowProxy> newDomWin =
       do_GetInterface(NS_ISUPPORTS_CAST(nsIBaseWindow*, window));
@@ -721,8 +708,10 @@
   nsCOMPtr<nsILoadContext> thisContext = do_GetInterface(newWebNav);
   if (thisContext) {
     thisContext->SetPrivateBrowsing(isPrivateBrowsingWindow);
-    thisContext->SetRemoteTabs(isUsingRemoteTabs);
-    thisContext->SetRemoteSubframes(isUsingRemoteSubframes);
+    thisContext->SetRemoteTabs(aChromeMask &
+                               nsIWebBrowserChrome::CHROME_REMOTE_WINDOW);
+    thisContext->SetRemoteSubframes(aChromeMask &
+                                    nsIWebBrowserChrome::CHROME_FISSION_WINDOW);
   }
 
   window.forget(aResult);
